src/diff/component.js

```diff
@@ -145,7 +147,7 @@ export function renderComponent(
 
 	if ((tmp = options._render)) tmp(internal);
 
-	c._dirty = false;
+	internal._mode &= ~DIRTY;
```

Comment:
Hmm I wonder if we could add `DIRTY` to `RESET_MODE`... technically it is a change in the existing behavior since we clear dirty before diffing children. Moving it to `RESET_MODE` would change it so we only clear dirty after diffing children. Perhaps for this PR we shouldn't change its behavior but perhaps it's something to consider for the future

Reply:
Tried that, but doing so breaks 42 tests :S

Reply:
lolol yea okay good call - let's not do that then